File: library/src/main/java/io/ashdavies/rx/tasks/SingleTaskListenerFactory.java
package io.ashdavies.rx.tasks;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import io.reactivex.SingleEmitter;

class SingleTaskListenerFactory<Result> implements TaskListenerFactory<Result, SingleEmitter<Result>> {

  @Override
  public OnSuccessListener<Result> createOnSuccessListener(SingleEmitter<Result> emitter) {
    return new SingleEmitterSuccessListener<>(emitter);
  }

  @Override
  public OnFailureListener createOnFailureListener(SingleEmitter<Result> emitter) {
    return new SingleEmitterFailureListener<>(emitter);
  }
}

File: library/src/main/kotlin/io/ashdavies/rx/rxtasks/SingleTaskListenerFactory.kt
package io.ashdavies.rx.rxtasks

import io.reactivex.Scheduler
import io.reactivex.SingleEmitter

internal class SingleTaskListenerFactory<T> constructor(override val scheduler: Scheduler? = null) : TaskListenerFactory<T, SingleEmitter<T>> {

  override fun createOnSuccessListener(emitter: SingleEmitter<T>) = SingleEmitterSuccessListener(emitter)

  override fun createOnFailureListener(emitter: SingleEmitter<T>) = SingleEmitterFailureListener(emitter)
}
